gamemodes/terrortown/gamemode/client/cl_search.lua

```diff
@@ -2,723 +2,393 @@
+	-- BUTTONS
+	local buttonArea = vgui.Create("DButtonPanelTTT2", frame)
+	buttonArea:SetSize(self.sizes.width, self.sizes.heightBottomButtonPanel)
+	buttonArea:Dock(BOTTOM)
+
+	local buttonReport = vgui.Create("DButtonTTT2", buttonArea)
+	buttonReport:SetText("search_call")
+	buttonReport:SetSize(self.sizes.widthButton, self.sizes.heightButton)
+	buttonReport:SetPos(0, self.sizes.padding + 1)
+	buttonReport.DoClick = function(btn)
+		bodysearch.ClientReportsCorpse(data.rag)
 	end
+	buttonReport:SetIcon(roles.DETECTIVE.iconMaterial, true, 16)
 
-	return bits
-end
-
-local search = {}
-
-local function TTTRagdollSearch()
-	search = {}
-
-	-- Basic info
-	search.eidx = net.ReadUInt(16)
-
-	local owner = Entity(net.ReadUInt(8))
-
-	search.owner = owner
-
-	if not IsValid(search.owner) or not search.owner:IsPlayer() or search.owner:IsTerror() then
-		search.owner = nil
+	if not bodysearch.CanReportBody(data.ragOwner) then
+		buttonReport:SetEnabled(false)
 	end
 
-	search.nick = net.ReadString()
-	search.role_color = net.ReadColor()
-
-	-- Equipment
-	local eq = {}
+	local playerCanTakeCredits = client:IsActiveShopper() and not clientRD.preventFindCredits
 
-	local eqAmount = net.ReadUInt(16)
-
-	for i = 1, eqAmount do
-		local eqStr = net.ReadString()
-
-		eq[i] = eqStr
-		search["eq_" .. eqStr] = true
-	end
+	local buttonConfirm = vgui.Create("DButtonTTT2", buttonArea)
 
-	-- Traitor things
-	search.role = net.ReadUInt(ROLE_BITS)
-	search.team = net.ReadString()
-	search.c4 = net.ReadInt(bitsRequired(C4_WIRE_COUNT) + 1)
-
-	-- Kill info
-	search.dmg = net.ReadUInt(30)
-	search.wep = net.ReadString()
-	search.head = net.ReadBit() == 1
-	search.dtime = net.ReadInt(16)
-	search.stime = net.ReadInt(16)
-
-	-- Players killed
-	local num_kills = net.ReadUInt(8)
-	if num_kills > 0 then
-		local t_kills = {}
-
-		for i = 1, num_kills do
-			t_kills[i] = net.ReadUInt(8)
+	if client:IsSpec() then
+		local text = "search_confirm"
+		if bodysearch.IsConfirmed(data.ragOwner) then
+			text = "search_confirmed"
 		end
 
-		search.kills = t_kills
+		buttonConfirm:SetEnabled(false)
+		buttonConfirm:SetText(text)
+		buttonConfirm:SetSize(self.sizes.widthButton, self.sizes.heightButton)
+		buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButton, self.sizes.padding + 1)
+	elseif bodysearch.IsConfirmed(data.ragOwner) then
+		buttonConfirm:SetText("search_confirmed")
+		buttonConfirm:SetEnabled(false)
+		buttonConfirm:SetSize(self.sizes.widthButton, self.sizes.heightButton)
+		buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButton, self.sizes.padding + 1)
+	elseif not bodysearch.CanConfirmBody() then
+		if data.credits > 0 and client:IsActiveShopper() and not clientRD.preventFindCredits then
+			if data.credits == 1 then
+				buttonConfirm:SetText("search_take_credit")
+				buttonConfirm:SetParams({credit = data.credits})
+			else
+				buttonConfirm:SetText("search_take_credits")
+				buttonConfirm:SetParams({credits = data.credits})
```

Comment:
I really dislike this style, because the value we gain is really near nothing and it is just more work for translators while also adding multiple extra if blocks. I feel like we should move towards simple and pragmatic solutions for such stuff, especially since no one has ever raised any concern regarding translations for such plural cases. 

What do you think?

Reply:
I would like that, if gmod had a proper translation library / system in place it would also have a pluralization library to handle this. 

We could investigate something like this in another PR in the future :)